src/main/java/org/entando/kubernetes/client/hub/DefaultHubClient.java

```diff
@@ -0,0 +1,114 @@
+                new ParameterizedTypeReference<PagedContent<BundleGroupVersionFilteredResponseView,
+                        BundleGroupVersionEntityDto>>() {
+                }, params);
+    }
+
+    @Override
+    public ProxiedPayload<PagedContent<BundleDto, BundleEntityDto>> getBundles(String host,
+            Map<String, Object> params) {
+        return doPagedGet(host, BUNDLE_API_PATH,
+                new ParameterizedTypeReference<PagedContent<BundleDto, BundleEntityDto>>() {
+                }, params);
+    }
+
+    protected <T> ProxiedPayload<T> doPagedGet(String host, String apiPath, ParameterizedTypeReference<? extends PagedContent> typedContent,
+            Map<String, Object> params) {
+        RestTemplate restTemplate = new RestTemplate();
+        ProxiedPayload<T> payload;
+
+        try {
+            UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(host);
+            generateUriBuilder(params, builder);
+            builder.path(apiPath);
+            final String endpointUrl = builder.build().toString();
+            ResponseEntity<? extends PagedContent> response = restTemplate.exchange(endpointUrl, HttpMethod.GET, null, typedContent);
+            payload = ProxiedPayload.<T>builder()
+                    .payload((T) response.getBody())
+                    .status(response.getStatusCode())
+                    .build();
+        } catch (RuntimeException t) {
+            log.error("error performing paged GET", t);
+            payload = ProxiedPayload.<T>builder()
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .exceptionMessage(t.getMessage())
+                    .exceptionClass(t.getClass().getCanonicalName())
+                    .build();
+        }
+        return payload;
+    }
+
+    protected <T> ProxiedPayload<T> doGet(String host, String apiPath, Map<String, Object> params) {
+        ProxiedPayload<T> payload;
+        RestTemplate restTemplate = new RestTemplate();
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(host);
+
+        try {
+            generateUriBuilder(params, builder);
+            builder.path(apiPath);
+            final String endpointUrl = builder.build().toString();
+            ResponseEntity<String> response
+                    = restTemplate.getForEntity(endpointUrl, String.class);
+            payload = ProxiedPayload.<T>builder()
+                    .payload((T) response.getBody())
+                    .status(response.getStatusCode())
+                    .build();
+        } catch (RuntimeException t) {
+            log.error("error performing paged GET", t);
+            payload = (ProxiedPayload<T>) ProxiedPayload.builder()
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .exceptionMessage(t.getMessage())
+                    .exceptionClass(t.getClass().getCanonicalName())
+                    .build();
+        }
+        return payload;
+    }
```

Comment:
Please delete this method since it seems to be never used and it is decreasing test coverage

Reply:
Done